virttest/qemu_devices/qcontainer.py

```diff
@@ -2259,7 +2259,12 @@ def serials_define_by_variables(self, serial_id, serial_type, chardev_id,
                                  parent_bus={'busid': bus}))
             devices[-1].set_param('name', serial_name)
         else:  # none virtio type, generate serial device directly
-            devices.append(qdevices.QDevice(serial_type, {"id": serial_id}))
+            if serial_type.startswith("pci"):
```

Comment:
If machine type is "Q35",  pci-serial will plug  under root port. is it expect ?

Reply:
> If machine type is "Q35", pci-serial will plug under root port. is it expect ?

@jingzhao84  can you have a try ?